gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/FsJobStatusRetriever.java

```diff
@@ -0,0 +1,163 @@
+   * @param flowGroup
+   * @return the latest flow execution id with the given flowName and flowGroup. -1 will be returned if no such execution found.
+   */
+  @Override
+  public long getLatestExecutionIdForFlow(String flowName, String flowGroup) {
+    Preconditions.checkArgument(flowName != null, "flowName cannot be null");
+    Preconditions.checkArgument(flowGroup != null, "flowGroup cannot be null");
+    try {
+      String storeName = Joiner.on(JobStatusRetriever.STATE_STORE_KEY_SEPARATION_CHARACTER).join(flowGroup, flowName);
+      List<String> tableNames = this.stateStore.getTableNames(storeName, input -> true);
+      if (tableNames.isEmpty()) {
+        return -1L;
+      }
+      Collections.sort(tableNames);
+      return getExecutionIdFromTableName(tableNames.get(tableNames.size() - 1));
+    } catch (Exception e) {
+      return -1L;
+    }
+  }
+
+  /**
+   *
+   * @param jobState instance of {@link State}
+   * @return deserialize {@link State} into a {@link JobStatus}.
+   */
+  private JobStatus getJobStatus(State jobState) {
+    String flowGroup = jobState.getProp(TimingEvent.FlowEventConstants.FLOW_GROUP_FIELD);
+    String flowName = jobState.getProp(TimingEvent.FlowEventConstants.FLOW_NAME_FIELD);
+    long flowExecutionId = Long.parseLong(jobState.getProp(TimingEvent.FlowEventConstants.FLOW_EXECUTION_ID_FIELD));
+    String jobName = jobState.getProp(TimingEvent.FlowEventConstants.JOB_NAME_FIELD);
+    String jobGroup = jobState.getProp(TimingEvent.FlowEventConstants.JOB_GROUP_FIELD);
+    long jobExecutionId = Long.parseLong(jobState.getProp(TimingEvent.FlowEventConstants.JOB_EXECUTION_ID_FIELD, "0"));
+    String eventName = jobState.getProp(JobStatusRetriever.EVENT_NAME_FIELD);
+    long startTime = Long.parseLong(jobState.getProp(TimingEvent.METADATA_START_TIME, "0"));
+    long endTime = Long.parseLong(jobState.getProp(TimingEvent.METADATA_END_TIME, "0"));
+    String message = jobState.getProp(TimingEvent.METADATA_MESSAGE, "");
+    String lowWatermark = jobState.getProp(TimingEvent.FlowEventConstants.LOW_WATERMARK_FIELD, "");
+    String highWatermark = jobState.getProp(TimingEvent.FlowEventConstants.HIGH_WATERMARK_FIELD, "");
+    long processedCount = Long.parseLong(jobState.getProp(TimingEvent.FlowEventConstants.PROCESSED_COUNT_FIELD, "0"));
+
+    return JobStatus.builder().flowName(flowName).flowGroup(flowGroup).flowExecutionId(flowExecutionId).
+        jobName(jobName).jobGroup(jobGroup).jobExecutionId(jobExecutionId).eventName(eventName).
+        lowWatermark(lowWatermark).highWatermark(highWatermark).startTime(startTime).endTime(endTime).
+        message(message).processedCount(processedCount).build();
+  }
+
+  private long getExecutionIdFromTableName(String tableName) {
+    return Long.parseLong(Splitter.on(JobStatusRetriever.STATE_STORE_KEY_SEPARATION_CHARACTER).splitToList(tableName).get(0));
+  }
+
+  private boolean shouldFilterJobStatus(List<String> tableNames, String tableName) {
```

Comment:
What is this filtering condition for? To filter jobs without a job group or name set?

Reply:
Exactly. Added javadoc to explain the filtering condition.